File: cpp/Elektronik.cpp
#include <iostream>
#include <string>
#include <iomanip> //librari biar bisa nampilin output nya bagus dan gapake angka ilmiah

using namespace std;

//deklarasi class
class Elektronik
{
    private:
        //private atribut
        string id_produk;
        string nama_produk;
        int stok;
        double harga;

    public:
    // constructor
    Elektronik(string id, string nama, int stok, double harga)
    {
        setId(id); //inisialisasi
        setNama(nama); //inisialisasi
        setStok(stok); //inisialisasi
        setHarga(harga); //inisialisasi
    }

    //setter
    void setId(const string& id) 
    {
        this->id_produk = id; //inisialisasi 
    }

    void setNama(const string& nama) 
    {
        this->nama_produk = nama; //inisialisasi
    }

    void setStok(const int& stok) 
    {
        this->stok = stok; //inisialisasi
    }

    void setHarga(const double& harga) 
    {
        this->harga = harga; //inisialisasi
    }


    // getter
    string getId() const
    {
        return id_produk; //mengambil value
    }

    string getNama() const
    {
        return nama_produk; //mengambil value
    }

    int getStok() const
    {
        return stok; //mengambil value
    }

    double getHarga() const
    {
        return harga; //mengambil value
    }
    
    //prosedur untuk menampilkan data
    void tampilkanData() const 
    {
        //print
        cout << "ID : " << getId() << endl
            << "Nama Produk : " << getNama() << endl
            << "Stok : " << getStok() << endl
            << "Harga : " << fixed << setprecision(0) << getHarga() << endl;
    }
    
    //destructor
    ~Elektronik()
    {
    }
};

File: php/Elektronik.php
<?php
    class Elektronik {
        //private atribut
        private string $id_produk;
        private string $nama_produk;
        private int $stok;
        private float $harga;
        private string $gambar;

        //constructor
        public function __construct(string $id_produk, string $nama_produk, int $stok, float $harga, string $gambar)
        {
            $this->id_produk = $id_produk; //inisialisasi atribut
            $this->nama_produk = $nama_produk; //inisialisasi atribut
            $this->stok = $stok; //inisialisasi atribut
            $this->harga = $harga; //inisialisasi atribut
            $this->gambar = $gambar; //inisialisasi atribut
        }

        // Getter (untuk mendapatkan nilai atribut)
        public function getId(): string
        {
            return $this->id_produk; //mengembalikan nilai atribut
        }

        public function getNama(): string
        {
            return $this->nama_produk; //mengembalikan nilai atribut
        }

        public function getStok(): int
        {
            return $this->stok; //mengembalikan nilai atribut
        }

        public function getHarga(): float
        {
            return $this->harga; //mengembalikan nilai atribut
        }

        public function getGambar(): string
        {
            return $this->gambar; //mengembalikan nilai atribut
        }

        // Setter (untuk mengubah nilai atribut)
        public function setId(string $id_produk): void
        {
            $this->id_produk = $id_produk; //menginisialisasi atribut dengan value baru
        }

        public function setNama(string $nama_produk): void
        {
            $this->nama_produk = $nama_produk; //menginisialisasi atribut dengan value baru
        }

        public function setStok(int $stok): void
        {
            // Validasi: memastikan stok tidak negatif
            if ($stok >= 0) {
                $this->stok = $stok; //menginisialisasi atribut dengan value baru
            } else {
                echo "Stok tidak boleh negatif."; //jika input tidak valid
            }
        }

        public function setHarga(float $harga): void
        {
            // Validasi: memastikan harga lebih dari 0
            if ($harga > 0) {
                $this->harga = $harga; //menginisialisasi atribut dengan value baru
            } else {
                echo "Harga harus lebih dari 0."; //jika input tidak valid
            }
        }

        public function setGambar(string $gambar): void
        {
            $this->gambar = $gambar; //menginisialisasi atribut dengan value baru
        }

        //function untuk menampilkan data
        public function tampilkanData(): void
        {
            //print
            echo "ID: " . $this->getIdProduk() . "<br>";
            echo "Nama Produk: " . $this->getNamaProduk() . "<br>";
            echo "Stok: " . $this->getStok() . "<br>";
            echo "Harga: " . $this->getHarga() . "<br>";
            echo "Gambar: " . $this->getGambar() . "<br>";
        }
    }
